extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php

```diff
@@ -1,17 +1,30 @@
 <?php
 
+use Wikia\PortableInfobox\Helpers\PagePropsProxy;
+use Wikia\PortableInfobox\Helpers\PortableInfoboxTemplatesHelper;
+use Wikia\PortableInfobox\Parser\Nodes\NodeInfobox;
+
 class PortableInfoboxDataService {
 
 	const IMAGE_FIELD_TYPE = 'image';
 	const INFOBOXES_PROPERTY_NAME = 'infoboxes';
 
-	/**
-	 * @var Title $title
-	 */
 	protected $title;
+	protected $templateHelper;
+	protected $propsProxy;
+	protected $cache;
+	protected $cachekey;
 
+	/**
+	 * @param $title Title
+	 *
+	 * @internal param $helper
+	 */
 	protected function __construct( $title ) {
-		$this->title = $title;
+		$this->title = $title !== null ? $title : new Title();
+		$this->templateHelper = new PortableInfoboxTemplatesHelper();
+		$this->propsProxy = new PagePropsProxy();
+		$this->cachekey = wfMemcKey( $this->title->getArticleID(), self::INFOBOXES_PROPERTY_NAME );
```

Comment:
Let's use a key prefix (e.g. `__CLASS__`) to make [memcache stats](https://github.com/Wikia/cheetah/blob/master/cheetah/reports/memcache_hit_ratio_report.py) a bit easier to read.